### ISSUE TITLE
Stricter prefer-const

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10499]
**What rule do you want to change?**

`prefer-const`

**Does this change cause the rule to produce more or fewer warnings?**

More.

**How will the change be implemented? (New option, new default behavior, etc.)?**

The default behavior should become stricter.

**Please provide some example code that this change will affect:**

```js
function test() {
  let foo = false;
  foo = false;
  return foo;
}
```

If a variable only uses primitives and these primitives are all identical it is likely a mistake. How exactly it should be fixed is something the developers should look into in such cases.

[Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiZnVuY3Rpb24gdGVzdCgpIHtcciAgbGV0IGZvbyA9IGZhbHNlO1xyICBmb28gPSBmYWxzZTtcciAgcmV0dXJuIGZvbztccn1cciIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OCwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJwcmVmZXItY29uc3QiOjJ9LCJlbnYiOnt9fX0=) of the current behavior.

**What does the rule currently do for this code?**

It does not warn.

**What will the rule do after it's changed?**

It would warn to change let to const. I am not sure how it would be best to warn about `foo = false`.

**Comment by not-an-aardvark** [comment]
This also seems loosely related to #8383.

I'm not sure about this because it seems like the case it's trying to detect (where the assignment value is specifically a primitive) is somewhat rare. Unfortunately, it's not possible to tell whether two arbitrary expressions will always have the same value, although it is possible to tell in certain special cases (e.g. when both expressions are literals). I suppose the rule could be worthwhile for catching errors in that case even though it doesn't occur frequently.

**Comment by BridgeAR** [comment]
@not-an-aardvark that would indeed fit together with your proposal.
